src/http.ts

```diff
@@ -148,18 +148,25 @@ export class LemmyHttp {
   #apiUrl: string;
   #headers: { [key: string]: string } = {};
   #pictrsUrl: string;
+  #fetchFunction = fetch;
 
   /**
    * Generates a new instance of LemmyHttp.
    * @param baseUrl the base url, without the vX version: https://lemmy.ml -> goes to https://lemmy.ml/api/vX
    * @param headers optional headers. Should contain `x-real-ip` and `x-forwarded-for` .
    */
-  constructor(baseUrl: string, headers?: { [key: string]: string }) {
+  constructor(
+    baseUrl: string,
+    options?: {
+      fetchFunction: typeof fetch;
```

Comment:
It looks like you don't do anything with the fetch function that is passed in. As it is now, this code will always use the default fetch function defined on line 151.

Reply:
oh wow, my comment didnt comment Dx

dunno how i missed that, fixed x3